src/PhpOption/Some.php

```diff
@@ -117,6 +117,34 @@ public function filterNot($callable)
         return None::create();
     }
 
+    public function filterIsA($class)
+    {
+        if (is_a($this->value, $class)) {
+            return $this;
+        }
+
+        return None::create();
+    }
+
+    public function filterIsOneOf(...$classes)
+    {
+        if (count($classes) < 1) {
+            return None::create();
+        }
+
+        if (is_array($classes[0])) {
```

Comment:
wouldn't you want to check for traversable?

Either way, seems you should do more of a (new Seq($classes))->flatten().

Reply:
traversable probably better.

no Seq, I don't think including another lib into Option makes sense just for this one use case. Plus I'm going to open this as a PR to the original and see if they will move on it (that repo has been inactive for 2 years)

Reply:
Too bad no php 7.1

http://php.net/manual/en/function.is-iterable.php

Reply:
yeah i intentionally avoided php7 isms (i.e. no return types...no argument types) since the base lib claims support for php5